src/components/reusable/daterangepicker/daterangepicker.ts

```diff
@@ -0,0 +1,288 @@
+      if (this.required && (!this.endDate || this.endDate === '')) {
+        this.internals.setValidity(
+          { valueMissing: true },
+          'Both dates are required.'
+        );
+        this.invalidText = this.internals.validationMessage;
+        return;
+      }
+      // validate min
+      if (this.endDate !== '' && this.minDate !== '') {
+        this.validateMinDate(this.endDate);
+      }
+      //validate max
+      if (this.endDate !== '' && this.maxDate !== '') {
+        this.validateMaxDate(this.endDate);
+      }
+      // validate start & end date
+      if (this.startDate !== '' && this.endDate !== '') {
+        this.validateStartEndDate();
+      }
+    }
+  }
+  // on-change start date
+  private handleStartDate(e: any) {
+    this.startDate = e.target.value;
+    if (this.startDate !== '') {
+      this.validateAndDispatchEvent();
+    }
+  }
+  // on-change end date
+  private handleEndDate(e: any) {
+    this.endDate = e.target.value;
+    if (this.endDate !== '') {
+      this.validateAndDispatchEvent();
+    }
+  }
+  // Note: dispatch (on-input) event only if both dates are valid (i.e. startDate <= endDate)
+  private validateAndDispatchEvent() {
+    if (this.startDate <= this.endDate) {
+      // emit selected start & end date value
+      const event = new CustomEvent('on-input', {
+        detail: {
+          startDate: this.startDate,
+          endDate: this.endDate,
+        },
+      });
+      console.log(`${event.detail.startDate} - ${event.detail.endDate}`);
+      this.dispatchEvent(event);
+    }
+  }
+  // validate minDate with start & end date
+  private validateMinDate(date: String): void {
+    if (this.regexDateFormat.test(this.minDate)) {
+      if (date < this.minDate) {
+        this.internals.setValidity(
+          { rangeUnderflow: true },
+          'Please enter date as min date or later.'
+        );
+        this.invalidText = this.internals.validationMessage;
+      }
+    } else {
+      this.internals.setValidity(
+        { patternMismatch: true },
+        'Please enter valid min date.'
+      );
+      this.invalidText = this.internals.validationMessage;
+    }
+  }
+  // validate maxDate with start & end date
+  private validateMaxDate(date: String): void {
+    if (this.regexDateFormat.test(this.maxDate)) {
+      if (date > this.maxDate) {
+        this.internals.setValidity(
+          { rangeOverflow: true },
+          'Please enter date as max date or earlier.'
+        );
+        this.invalidText = this.internals.validationMessage;
+      }
+    } else {
+      this.internals.setValidity(
+        { patternMismatch: true },
+        'Please enter valid max date.'
+      );
+      this.invalidText = this.internals.validationMessage;
+    }
+  }
+
+  private validateStartEndDate(): void {
+    if (this.startDate > this.endDate) {
+      this.internals.setValidity(
+        { patternMismatch: true },
+        'Please enter valid start date & end date.'
```

Comment:
Maybe change this message to something like: "State date must be before end date."

Reply:
Done. Only one doubt here. can start date and end date be same ? currently I allowed this scenario.

 
<img width="423" alt="Screenshot 2023-09-08 at 11 59 55 PM" src="https://github.com/kyndryl-design-system/web-components/assets/140634142/ee20d86e-4e83-406f-8c6f-ee241b3968dd">

Reply:
Good question. but I think it's OK as is. I had the same thought during initial review.

Reply:
ok